src/pages/Swap/index.tsx

```diff
@@ -558,7 +558,7 @@ export function Swap({
         showCancel={true}
       />
       <SwapHeader trade={trade} autoSlippage={autoSlippage} chainId={chainId} />
-      {trade && showConfirm && (
+      {trade && showConfirm && allowance.state !== AllowanceState.LOADING && (
```

Comment:
@just-toby since the ui gets a response so quickly from the mocked quote endpoint, confirm swap was sometimes getting clicked before allowances had been loaded in, skipping the approval step of the flow. I added in this check to prevent that, just want to check w/ you that this seems like the right place to do it

Reply:
does `allowance` ever reload after the modal is already open? if we're refreshing the allowance and it's `LOADING` during the flow maybe the modal will auto-close...

Reply:
just tested and it didn't happen